src/go/rpk/pkg/cli/cluster/config/set.go

```diff
@@ -60,92 +78,152 @@ You may also use <key>=<value> notation for setting configuration properties:
 If an empty string is given as the value, the property is reset to its default.`,
 		Args: cobra.MinimumNArgs(1),
 		Run: func(cmd *cobra.Command, args []string) {
-			var key, value string
-			if len(args) == 1 && strings.Contains(args[0], "=") {
-				kv := strings.SplitN(args[0], "=", 2)
-				key, value = kv[0], kv[1]
-			} else if len(args) == 2 {
-				key, value = args[0], args[1]
+			configs, err := parseArgs(args)
+			out.MaybeDieErr(err)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			vp := cfg.VirtualProfile()
+
+			if vp.FromCloud {
+				if !vp.CloudCluster.IsBYOC() {
```

Comment:
Should also be supported for dedicated clusters

Reply:
👍 changed this validation only for serverless